tfhe/src/integer/gpu/server_key/radix/shift.rs

```diff
@@ -0,0 +1,402 @@
+                    self.message_modulus,
+                    self.carry_modulus,
+                    d_multibit_bsk.glwe_dimension,
+                    d_multibit_bsk.polynomial_size,
+                    self.key_switching_key
+                        .input_key_lwe_size()
+                        .to_lwe_dimension(),
+                    self.key_switching_key
+                        .output_key_lwe_size()
+                        .to_lwe_dimension(),
+                    self.key_switching_key.decomposition_level_count(),
+                    self.key_switching_key.decomposition_base_log(),
+                    d_multibit_bsk.decomp_level_count,
+                    d_multibit_bsk.decomp_base_log,
+                    d_multibit_bsk.grouping_factor,
+                    lwe_ciphertext_count.0 as u32,
+                    is_signed,
+                );
+            }
+        }
+    }
+
+    /// # Safety
+    ///
+    /// - `stream` __must__ be synchronized to guarantee computation has finished, and inputs must
+    ///   not be dropped until stream is synchronised
+    pub unsafe fn unchecked_right_shift_async<T>(
+        &self,
+        ct: &T,
+        shift: &CudaUnsignedRadixCiphertext,
+        stream: &CudaStream,
+    ) -> T
+    where
+        T: CudaIntegerRadixCiphertext,
+    {
+        let mut result = ct.duplicate_async(stream);
+        self.unchecked_right_shift_assign_async(&mut result, shift, stream);
+        result
+    }
+
+    /// Computes homomorphically a right shift by an encrypted amount
+    ///
+    /// The result is returned as a new ciphertext.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use tfhe::core_crypto::gpu::{CudaDevice, CudaStream};
+    /// use tfhe::integer::gpu::ciphertext::CudaUnsignedRadixCiphertext;
+    /// use tfhe::integer::gpu::gen_keys_radix_gpu;
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    ///
+    /// let gpu_index = 0;
+    /// let device = CudaDevice::new(gpu_index);
+    /// let mut stream = CudaStream::new_unchecked(device);
+    ///
+    /// let size = 4;
+    /// // Generate the client key and the server key:
+    /// let (cks, sks) = gen_keys_radix_gpu(PARAM_MESSAGE_2_CARRY_2_KS_PBS, size, &mut stream);
+    ///
+    /// let msg = 128;
+    /// let shift = 2;
+    ///
+    /// let ct = cks.encrypt(msg);
+    /// let shift_ct = cks.encrypt(shift as u64);
+    /// // Copy to GPU
+    /// let mut d_ct = CudaUnsignedRadixCiphertext::from_radix_ciphertext(&ct, &mut stream);
+    /// let mut d_shift_ct = CudaUnsignedRadixCiphertext::from_radix_ciphertext(&shift_ct, &mut stream);
+    ///
+    /// let d_ct_res = sks.unchecked_right_shift(&d_ct, &d_shift_ct, &mut stream);
+    ///
+    /// // Copy back to CPU
+    /// let ct_res = d_ct_res.to_radix_ciphertext(&mut stream);
+    ///
+    /// // Decrypt:
+    /// let dec_result: u64 = cks.decrypt(&ct_res);
+    /// assert_eq!(dec_result, msg >> shift);
+    /// ```
```

Comment:
Same comment for this file, it would be good to move the doc tests on the default entry points.